### DISCUSSION TITLE
Improvement of comparison operator

### DISCUSSION THREAD

**Discussion by andrey-messi** [dotnet/csharplang#1128] [General]
```
/// <summary>
/// Check if var2 exceeds var1 and they don't exceed bounds
/// </summary>
public static bool Check(int var1, int var2)
{
	int maxLimit = 10;
	int minLimitStrict = 1;
	return (minLimitStrict < var2 && var2 < var1 && var1 <= maxLimit);
}
```
It would be more convinient to use
`(minLimitStrict < var2 < var1 <= maxLimit)`
instead of
`(minLimitStrict < var2 && var2 < var1 && var1 <= maxLimit)`

I know that here a compiler firstly evaluates first comparison operator (minLimitStrict < var2) and returns bool, so code:
`(minLimitStrict < var2 < var1 <= maxLimit)`
is not valid now.
But maybe it is possible to change a compiler's behavior and obligate to use parentheses when expected to return bool. 
Or if it is not possible (for example for not breaking existing thirt-party code base), then it will be good to introduce new operators, for example <* and <=* :
`(minLimitStrict <* var2 <* var1 <=* maxLimit)`

**Comment by ufcpp** [comment]
dup of https://github.com/dotnet/csharplang/issues/217 https://github.com/dotnet/csharplang/issues/847
related to https://github.com/dotnet/csharplang/issues/420

**Comment by jnm2** [comment]
Honestly if I had a dollar for every time I could have used this in C#, I'd be making about $3/year.

**Comment by bondsbw** [comment]
>Honestly if I had a dollar for every time I could have used this in C#, I'd be making about $3/year.

If you've been using C# since the beginning, then you might have used this 50 times already.  There are several "really useful" C# features I've used less than that over my career.
